src/tdlib/chat.rs

```diff
@@ -308,32 +253,35 @@ glib::wrapper! {
 }
 
 impl Chat {
-    pub(crate) fn new(chat: TelegramChat, session: Session) -> Self {
-        let type_ = ChatType::from_td_object(&chat.r#type, &session);
-        let avatar = chat.photo.map(Avatar::from);
-        let draft_message = chat.draft_message.map(BoxedDraftMessage);
-
-        glib::Object::new(&[
-            ("id", &chat.id),
-            ("type", &type_),
-            ("is-blocked", &chat.is_blocked),
-            ("title", &chat.title),
-            (
-                "last-read-outbox-message-id",
-                &chat.last_read_outbox_message_id,
-            ),
-            ("avatar", &avatar),
-            ("draft-message", &draft_message),
-            ("unread-mention-count", &chat.unread_mention_count),
-            ("unread-count", &chat.unread_count),
-            (
-                "notification-settings",
-                &BoxedChatNotificationSettings(chat.notification_settings),
-            ),
-            ("permissions", &BoxedChatPermissions(chat.permissions)),
-            ("session", &session),
-        ])
-        .expect("Failed to create Chat")
+    pub(crate) fn new(td_chat: TelegramChat, session: &Session) -> Self {
+        let chat: Chat = glib::Object::new(&[]).expect("Failed to create Chat");
+        let imp = chat.imp();
+
+        let type_ = ChatType::from_td_object(&td_chat.r#type, &session);
+        let avatar = td_chat.photo.map(Avatar::from);
+        let draft_message = td_chat.draft_message.map(BoxedDraftMessage);
+        let notification_settings = BoxedChatNotificationSettings(td_chat.notification_settings);
+        let permissions = BoxedChatPermissions(td_chat.permissions);
+
+        imp.id.set(td_chat.id);
+        imp.type_.set(type_).unwrap();
+        imp.is_blocked.set(td_chat.is_blocked);
+        imp.title.replace(td_chat.title);
+        imp.avatar.replace(avatar);
+        imp.last_read_outbox_message_id
+            .set(td_chat.last_read_outbox_message_id);
+        // TODO: Initialize last-message
+        // TODO: Initialize order
+        // TODO: Initialize is-pinned
```

Comment:
Forgotten?

Reply:
Not really since they were not initialized before, but actually it makes sense to add a commit to actually initialize them now.